### DISCUSSION TITLE
Duplicate a User Role

### DISCUSSION THREAD

**Discussion by zeusstl** [directus/directus#2781] [Feature Requests]
Add a button on the user role page that allows you to duplicate the current user role.

Why is this important?

1. More often than not, when you are creating a new user role, it will share a lot in common with an existing user role. Like an employee might be very similar to a manager except for a few minor differences. Or a member and visitor might be the same besides a few differences.
2. User Roles have a lot of detail, so it is a pain to replicate from one role to the other.

(For super bonus points, add the ability to transfer permissions from one role to another AND/OR have permission groupings that can be added to or applied to a role. Or some concept of inheritance where a child role can override or build on top of a parent role.)

I'm overloaded and can't get to this myself, yet, but wanted the idea out in the world in case someone else has the urge and time.

**Comment by fracosfr** [comment]
Hi! 

If I can suggest a secondary function about the roles : It will be great if the roles can inherit from other roles ;)

**Reply by rijkvanzanten** [reply]
>  It will be great if the roles can inherit from other roles ;)

Mind opening a new feature request for that? I'm curious to hear more detail on how you think it would ideally work (for example, would permissions be merged on an "AND" or "OR" basis?)

**Reply by fracosfr** [reply]
It's done, with captures => https://github.com/directus/directus/discussions/15347

**Comment by nicksergeant** [comment]
🤔 this discussion is marked as "Implemented" but I can't for the life of me figure out where / how I can duplicate a user role in Directus admin... does anyone know where I can find this?
